code/game/objects/items/RPD.dm

```diff
@@ -479,7 +479,8 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 					if (target_differences & S.connections)
 						to_chat(user, span_warning("\The [src]'s screen flashes a warning: Can't configure a pipe in a currently connected direction."))
 						return
-					var/new_dir = (S.GetInitDirections() & ~target_differences) | target_differences
+					var/old_dir = S.GetInitDirections()
+					var/new_dir = (old_dir & ~target_differences) | (target_init_dir & target_differences)
```

Comment:
isn't `target_init_dir` already `S.GetInitDirections()`? What's the actual goal of this line?

Reply:
Good catch! The line attempts to calculate the new direction restrictions. To do this, it should retain restrictions that aren't changing (`old_dir & ~target_differences`) and set desired bits from what we are changing (this should be `p_init_dir & target_differences` rather than `target_init_dir & target_differences`).

Reprogramming in general exists to solve two cases:

- Consider a smart pipe that is currently limited to connecting horizontally, with an scrubber immediately below it that wants to connect upwards. The smart pipe isn't allowed to connect downwards, so will not connect to the scrubber. The reprogramming functionality exists to let a smart pipe be reprogrammed so it can connect downwards, without needing to remove the pipe and place a less restricted one.
- Consider a smart pipe that is very permissive, willing to connect in all directions. We want to make a pipeline immediately parallel, without connecting to the permissive pipe. We can reprogram the overly permissive pipe to not be allowed to connect in this direction, without needing to remove the pipe and place a more restrictive one.

The old line always allowed everything out of what's changing, and the new line doesn't change anything. I'm not sure how this ended up being the PR'd version, but replacing `target_init_dir` with `p_init_dir` should be sufficient. I'll probably add in a new comment to be clear.